applications/nrf5340_audio/src/bluetooth/bt_stream/broadcast/broadcast_sink.c

```diff
@@ -343,6 +343,12 @@ static void syncable_cb(struct bt_bap_broadcast_sink *sink, bool encrypted)
 
 	LOG_DBG("Broadcast sink is syncable");
 
+	if (sink != broadcast_sink) {
+		broadcast_sink_cleanup();
```

Comment:
Wouldn't it be better to do the `broadcast_sink_cleanup` where you create the new broadcast sink instead? So there is only ever a single broadcast sink object?

Reply:
We do that if we have an ongoing stream and then suddenly call the sink create, I guess we could just run cleanup on every pa_sync_set (before create) then

Reply:
The problem with this approach is that you will overwrite `broadcast_sink` on the second create and lose the reference you had on the first, and thus not being able to delete that later. 

In other words, you are going to have a memory leak if you only have a single global `broadcast_sink` pointer that you overwrite with multiple calls to `bt_bap_broadcast_sink_create`.